orchestrator/careplancontributor/service.go

```diff
@@ -117,6 +126,86 @@ func (s Service) handleProxyToEPD(writer http.ResponseWriter, request *http.Requ
 	proxy.ServeHTTP(writer, request)
 }
 
+func (s Service) handleProxyToFHIR(writer http.ResponseWriter, request *http.Request) error {
+	// Authorize requester before proxying FHIR request
+	// Data holder must verify that the requester is part of the CareTeam by checking the URA
+	// Validate by retrieving the CarePlan from CPS, use URA in provided token to validate against CareTeam
+	// CarePlan should be provided in X-Scp-Context header
+	carePlanURLValue := request.Header[carePlanURLHeaderKey]
+	if len(carePlanURLValue) == 0 {
+		return errors.New(fmt.Sprintf("%s header value must be set", carePlanURLHeaderKey))
+	}
+	if len(carePlanURLValue) != 1 {
+		return errors.New(fmt.Sprintf("%s header must only contain one value", carePlanURLHeaderKey))
+	}
+	carePlanURL := carePlanURLValue[0]
+	if carePlanURL == "" {
+		return errors.New(fmt.Sprintf("%s header value must be set", carePlanURLHeaderKey))
+	}
+	if !strings.HasPrefix(carePlanURL, s.carePlanServiceURL.String()) {
+		return errors.New("invalid CarePlan URL in header")
+	}
+	u, err := url.Parse(carePlanURL)
+	if err != nil {
+		return err
+	}
+
+	carePlanServiceClient := fhirclient.New(s.carePlanServiceURL, s.scpHttpClient, coolfhir.Config())
+	var bundle fhir.Bundle
+	// Use extract CarePlan ID to be used for our query that will get the CarePlan and CareTeam in a bundle
+	carePlanId := strings.TrimPrefix(strings.TrimPrefix(u.Path, "/fhir/CarePlan/"), s.carePlanServiceURL.String())
+	newUrl := fmt.Sprintf("CarePlan?_id=%s&_include=CarePlan:care-team", carePlanId)
+	err = carePlanServiceClient.Read(newUrl, &bundle)
+	if err != nil {
+		return err
+	}
+	// We are expecting a total of 1 result, with 2 entries (one for CarePlan, one for CareTeam)
+	if *bundle.Total == 0 {
```

Comment:
wouldn't checking `len(bundle.Entry)` be safer? (and constant with the check below)

And I think if we leave out the 2 checks , the following check would stil fail right:

```
	if len(careTeams) == 0 {
		return errors.New("CareTeam not found in bundle")
	}
```

so do we need them? Or do we need to reassure the CarePlan exists (although I don't really see that being explciti here).

Reply:
I can see an error for the CarePlan not existing being useful as an error to return to the user so I will do an explicit check for that, but otherwise yes I think the other checks can be removed to streamline the validation.

Not sure why I checked bundle.Total... Definitely not consistent with the rest of the validation